### PR TITLE
wip/6.0 Porting legacy 'entity graph' tests

### PR DESCRIPTION
**Pull request by NathanQingyangXu**
Majority of legacy testing cases are enabled, though a few cannot be done for `NotYetImplementedFor6Exception` exceptions. 
Some major coding bugs are flushed out.
Legacy testing cases are modernized as much as possible during the importing to enhance their maintainability.
A logic bug in `InitializationCheckMatcher` is corrected as well. See self-comment for details.

The good news is the entity graph implementation seems working fine by passing the legacy testing cases.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java [hunk 1]
```diff
@@ -106,10 +107,10 @@ public StandardSqmSelectTranslator(
...
 		super( creationContext, queryOptions, fetchInfluencers, domainParameterXref, domainParameterBindings );
 
-		if ( fetchInfluencers != null
-				&& fetchInfluencers.getEffectiveEntityGraph() != null
-				&& fetchInfluencers.getEffectiveEntityGraph().getSemantic() != null ) {
-			this.entityGraphTraversalState = new StandardEntityGraphTraversalStateImpl( fetchInfluencers.getEffectiveEntityGraph() );
+		final AppliedGraph appliedGraph = queryOptions.getAppliedGraph();
+		if (  appliedGraph.getSemantic() != null && appliedGraph.getGraph() != null ) {
+			this.entityGraphTraversalState = new StandardEntityGraphTraversalStateImpl(
+					appliedGraph.getSemantic(), appliedGraph.getGraph() );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java:113]: This is the biggest coding bug flushed out. For HQL/Criteria, we should grab entity graph info from `QueryOptions`, not `LoadQueryInfluencers`.

File: hibernate-testing/src/main/java/org/hibernate/testing/hamcrest/InitializationCheckMatcher.java [hunk 2]
```diff
@@ -42,12 +42,16 @@ public InitializationCheckMatcher(boolean assertInitialized) {
...
 	public boolean matches(Object item) {
-		final boolean initialized = Hibernate.isInitialized( item );
-		return assertInitialized ? initialized : !initialized;
+		return assertInitialized == Hibernate.isInitialized( item );
 	}
 
 	@Override
 	public void describeTo(Description description) {
-		description.appendValue( "Hibernate#isInitialized == " + !assertInitialized );
+		description.appendValue( "Hibernate#isInitialized() returns " + assertInitialized );
```
Comment (NathanQingyangXu) [inline hibernate-testing/src/main/java/org/hibernate/testing/hamcrest/InitializationCheckMatcher.java:50]: The above method will be used for 'Expected', so the original code logic is incorrect. The 'actual' counterpart is customized below.

File: hibernate-core/src/test/java/org/hibernate/orm/test/entitygraph/parser/EntityGraphParserTest.java [hunk 3]
```diff
@@ -262,7 +244,6 @@ private void checkMapKeyAndValueSubgraphs(EntityGraph<GraphParsingTestEntity> gr
 	}
 
 	@Test
-//	@Ignore("Cannot run due to Hibernate bug: https://hibernate.atlassian.net/browse/HHH-12696")
 	public void testHHH12696MapSubgraphsValueFirst() {
 		EntityManager entityManager = getOrCreateEntityManager();
 		EntityGraph<GraphParsingTestEntity> graph = entityManager.createEntityGraph( GraphParsingTestEntity.class );
```
Comment (NathanQingyangXu) [inline hibernate-core/src/test/java/org/hibernate/orm/test/entitygraph/parser/EntityGraphParserTest.java:249]: Both HHH-12696 and HHH-10378 have been fixed already.

File: hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java [hunk 4]
```diff
@@ -473,25 +483,30 @@ private void applyOrdering(TableGroup tableGroup, OrderByFragment orderByFragmen
...
-					final String fetchParentEntityName = entityMappingType.getEntityName();
-					final String fetchableRole = fetchParentEntityName + "." + fetchable.getFetchableName();
-
-					for ( String enabledFetchProfileName : loadQueryInfluencers.getEnabledFetchProfileNames() ) {
-						final FetchProfile enabledFetchProfile = creationContext.getSessionFactory().getFetchProfile( enabledFetchProfileName );
-						final org.hibernate.engine.profile.Fetch profileFetch = enabledFetchProfile.getFetchByRole( fetchableRole );
-
-						fetchTiming = FetchTiming.IMMEDIATE;
-						joined = joined || profileFetch.getStyle() == org.hibernate.engine.profile.Fetch.Style.JOIN;
+			if ( ! (fetchable instanceof CollectionPart ) ) {
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java:486]: This is another big coding bug flushed out. We are supposed to skip entity graph traversal for collection element fetchable or map index fetchable. Traversal has been done already during the collection fetchable processing.

**Comment by dreab8** [comment]
merged, thank @NathanQingyangXu

**Pull request closed without merging**